typings/xterm.d.ts

```diff
@@ -839,6 +839,12 @@ declare module 'xterm' {
      */
     onRender: IEvent<{ start: number, end: number }>;
 
+    /**
+     * Adds an event listener for when a chunk of data has been
+     * processed in the write buffer.
+     */
+    onBufferContentsChange: IEvent<void>;
```

Comment:
Was just writing up some jsdoc to also call out the fact that this isn't entirely correct. I think the better thing here is to make the event name more correct so we don't need to call out caveats.

```suggestion
    /**
     * Adds an event listener for when data has been parsed by the terminal,
     * after {@link write} is called. This event is useful to listen for any
     * changes in the buffer.
     * 
     * This fires at most once per frame, after data parsing completes. Note
     * that this can fire when there are still writes pending if there is a lot
     * of data.
     */
    onWriteParsed: IEvent<void>;
```